source/develop/release-management/features/network/liveMigrationSupportForSRIOV.html.md

```diff
@@ -0,0 +1,83 @@
+---
+title: LiveMigrationSupportForSRIOV
+category: feature
+authors: mmucha
+wiki_category: Feature
+wiki_title: Features/LiveMigrationSupportForSRIOV
+wiki_revision_count: 12
+wiki_last_updated: 2015-10-14
+feature_name: Live Migration Support For SRIOV
+feature_modules: Networking
+feature_status: Design
+rfe: https://bugzilla.redhat.com/show_bug.cgi?id=868811
+
+---
+
+
+# Live Migration Support For SRIOV
+
+### Owner
+
+*   Name: [ Martin Mucha](User:mmucha)
+*   Email: mmucha@redhat.com
+
+## Summary
+Current support of SR-IOV in engine does not allow migration, which
+limits its usability. However current support of SR-IOV does support 
+hot-plugging and hot-unplugging. We can employ this to enable currently 
+not supported migration. When VM is migrated to another host, its
+passthrough nic will be hot-unplugged, and related VF will be released.
+Then we can perform VM migration, and after that we'll perform hot-plug.
+Therefore, after migrating VM, there will be slight delay before nic 
+reappears in VM. 
+
+## Detailed Description
+
+Currently, VMs using SR-IOV nics cannot be migrate. To preserve this 
+behavior user can specify, whether each passthrough nic is 'migratable' 
+or not. VM will can only be migrated, when all its nics passthrough 
+nics are marked as migratable. 
+
+Hot-plug can succeed only if there's available virtual function on 
+target host. To avoid possible race, migration will allocate virtual 
+function first, proceeding only if there was available one. 
+If there's none, VM won't be migrated.
+
+Migration with SR-IOV vNICs is a tricky multi-state operation, and can 
+fail. If the operation fails after the VM is already running on the 
+destination, the VM would not be migrated back.
+
+### REST
+
+Model will be altered, so that vNicProfile can be set as migratable:
+```
+@Type
+public interface VnicProfile extends Identified {
+    …
+    VnicPassThrough passThrough();
+    boolean migratable();
+```
+
+### GUI
+
+In UI, you need to flag passthrough nic as migratable to be able to do
+migration when SR-IOV is used. Only passthrough nic can be marked as
+migratable, or not marked as migratable. Other nics are always 
+migratable, thus migratable checkbox will be selected and greyed out
+when passthrough checkbox is not selected. 
+
+#### Setting migratable flag
+![Vnic profile with migratable flag png](vnicProfileWithMigratableFlag.png "Vnic profile with migratable flag png")
+
+###Guest-side support.
+   
+While migrating, the guest can notice that one of its vNICs has been 
```

Comment:
Here we ask our users to define an SRIOV network and an equivalent bridged network; to attach both to the VM; to allow mac spoofing; and to configure an in-guest bonding.

@mykaul wants us to put less burden on our users. Can you think of a grander approach, in which a user defines a "migratable passthrough network", and when attaching it to a VM, **two** vNICs are created, filter is set, and the guest is instructed to bond the two? It does not seem very easy; but maybe you can do it as a second phase of the feature as it is currently defined?